draft-ietf-ohai-ohttp.md

```diff
@@ -974,12 +976,23 @@ examine the content of a request (other than to observe the choice of key
 identifier, KDF, and AEAD), so it is also assumed that it cannot generate an
 encapsulated response.
 
-A proxy MUST NOT add information about the client identity when forwarding
-requests. This includes the Via field, the Forwarded field
-{{?FORWARDED=RFC7239}}, and any similar information.  A client does not depend
-on the proxy using an authenticated and encrypted connection to the oblivious
-request resource, only that information about the client not be attached to
-forwarded requests.
+A proxy MUST NOT reveal information about the client identity when forwarding
+requests without agreement from the client, with one exception (described
+below). For example, including the Via field, the Forwarded field
+{{?FORWARDED=RFC7239}}, or similar information may reveal information about the
+client that the proxy is intended to hide from the target. In general, a client
+does not depend on the proxy using an authenticated and encrypted connection to
+the oblivious request resource, only that information about the client not be
+attached to forwarded requests.
+
+The only exception to this requirement is that a proxy MAY add at most a single
```

Comment:
Hmm, I'm not convinced that this helps. I think if a proxy is interested in adding metadata to prevent fraud, then it would be unlikely that a single bit would suffice.
I'm personally in favour of changing the "MUST NOT add information about a client" to a SHOULD NOT, and craft verbiage to outline that this is a bad idea for fingerprinting/loss of anonymity etc reasons - basically considerations that a proxy should take into account.

Reply:
I propose we separate these two issues: (1) what information is the proxy allowed to attach (or not), and (2) for information it's allowed to attach, how much? I think we can reasonably address (1) in this PR, whereas (2) requires more analysis (or a lot more language). Would that work?

Reply:
I'm okay with that split, but for (1), were you thinking specific use-cases within antifraud, or saying "proxies should not attach metadata except for antifraud reasons because of XYZ reasons", or something else?

Reply:
I don't think we (even outside of OHAI) have consensus on what is suitable for anti-abuse purposes, so this PR is mainly trying to set up the draft to have that discussion separately. In other words, this PR is just about saying proxies MUST NOT attach data the client hasn't agreed to being revealed, except for this single bit. We can then bike shed the number of bits separately.

Reply:
I think exactly because of that lack of consensus the text should be a SHOULD! I'm also okay with saying "MUST NOT except for anti-fraud purposes" and then leaving how many bits that needs etc to a separate discussion, but IMO "MUST NOT... except" sounds a lot like SHOULD.

Reply:
Okay, I'm good with that framing in this PR if we exclude discussion about bits etc. I thought you were suggesting still including that when you said
>In other words, this PR is just about saying proxies MUST NOT attach data the client hasn't agreed to being revealed, except for this single bit.

Reply:
I think we're still not aligned :-) The purpose of this PR is to allocate just a single bit for shadowban or whatever. Future PRs can relax this constraint and allow more bits, subject to more analysis or whatever's needed. I don't think we should have text that says "proxies MAY attach TBD bits [sort out details later]." Does that clarify?

Reply:
Gotcha, I'm fine with that.